MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs

```diff
@@ -1,1637 +1,2306 @@
-                finally
-                {
-                    onedNodeMaskPinned.Free();
-                    api.DeallocateState(id);
-                }
-            }
-        }
-
-        [Test]
-        public void GetSelectedVerticesInPolygonThroughAPI()
-        {
-            using (var mesh = GenerateRegularGrid(4, 4, 1, 1))
-            using (var api = new MeshKernelApi())
-            {
-                var id = 0;
-                try
-                {
-                    id = api.AllocateState(0);
-
-                    Assert.IsTrue(api.Mesh2dSet(id, mesh));
-
-                    var geometryListIn = new DisposableGeometryList();
-
-                    geometryListIn.XCoordinates = new[] { 1.5, 1.5, 3.5, 3.5, 1.5 };
-                    geometryListIn.YCoordinates = new[] { -1.5, 1.5, 1.5, -1.5, -1.5 };
-                    geometryListIn.Values = new[] { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 };
-                    geometryListIn.NumberOfCoordinates = geometryListIn.XCoordinates.Length;
-                    geometryListIn.GeometrySeparator = api.GetSeparator();
-
-                    var selectedVertices = api.GetSelectedVerticesInPolygon(id, ref geometryListIn, 1);
-                    Assert.AreEqual(selectedVertices.Length, 4);
-                }
-                finally
-                {
-                    api.DeallocateState(id);
-                }
-            }
-        }
-    }
-}
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using MeshKernelNETCore.Api;
+using NUnit.Framework;
+using static MeshKernelNETCoreTest.Api.TestUtilityFunctions;
+
+namespace MeshKernelNETCoreTest.Api
+{
+    [TestFixture]
+    [Category("MeshKernelNETTests")]
+    public class MeshKernelTest
+    {
+        [Test]
+        [Category("PerformanceTests")]
+        public void Mesh2dDeleteNodeThroughApiPerformanceTrace()
+        {
+            // Setup
+            using (var api = new MeshKernelApi())
+            using (var mesh = CreateMesh2D(100, 100, 100, 200))
+            {
+                var stopWatch = new Stopwatch();
+                var numberOfVerticesBefore = mesh.NumNodes;
+                var id = 0;
+                GetTiming(stopWatch, "Create grid state", () =>
+                {
+                    id = api.AllocateState(0);
+
+                });
+
+                GetTiming(stopWatch, "Set state", () => { Assert.IsTrue(api.Mesh2dSet(id, mesh)); });
+
+                GetTiming(stopWatch, "Delete node", () => { Assert.IsTrue(api.Mesh2dDeleteNode(id, 0)); });
+
+                GetTiming(stopWatch, "Get mesh state", () =>
+                {
+                    var mesh2d = new DisposableMesh2D();
+                    var success = api.Mesh2dGetData(id, out mesh2d);
+                    Assert.True(success);
+
+                    var count = mesh2d.NodeX.Length;
+                    Assert.AreEqual(numberOfVerticesBefore - 1, count);
+
+                    mesh2d.Dispose();

```

Comment:
I would not do the call to dispose in the performance measure. This can be moved outside of the action.